lib/mayaUsd/ufe/UsdAttributes.cpp

```diff
@@ -124,22 +129,23 @@ Ufe::Attribute::Type UsdAttributes::attributeType(const std::string& name)
     }
 #ifdef UFE_V4_FEATURES_AVAILABLE
     Ufe::NodeDef::Ptr nodeDef = UsdAttributes::nodeDef();
-    if (!nodeDef) {
-        return Ufe::Attribute::kInvalid;
-    }
-    Ufe::AttributeDef::ConstPtr attrDef = nameToAttrDef(tok, nodeDef);
-    if (attrDef) {
-        return attrDef->type();
+    if (nodeDef) {
+        Ufe::AttributeDef::ConstPtr attrDef = nameToAttrDef(tok, nodeDef);
+        if (attrDef) {
+            return attrDef->type();
+        }
     }
 #endif
-    return Ufe::Attribute::kInvalid;
+    std::string msg = "Can not find attribute \"" + name + "\" on scene item \""
+        + fItem->path().string() + "\".";
+    throw std::invalid_argument(msg);
 }
 
 Ufe::Attribute::Ptr UsdAttributes::attribute(const std::string& name)
 {
     // early return if name is empty.
     if (name.empty()) {
-        return nullptr;
```

Comment:
The main Python call site is in UFE itself. It is one that I can actually point to on this public forum since it is installed in Maya.
Look for `/runTime/Python/Lib/site-packages/ufe/__init__.py`
When passing an empty string, or a name that does not exist, the code in `typedAttributeFactory()` calls `attrs.attributeType(attrName)`, receives `Unknown` then complains that this is not found in `typedAttributeDict`. I found that to be a bit too obscure, hence the exception. I would have preferred `KeyError` instead of `ValueError` on the Python side though.

Reply:
I do have to agree that switching error handling on short notice is bug-prone though, so I would have no issue reverting that change.